### PR TITLE
remove save button for consistant/clean editor behaviour

### PR DESCRIPTION
**Pull request by Mayo19**

### CODE REVIEW

**Comment by Mayo19** [comment]
solves the problem in this gif (in the link) by removing the additional save button. works just like the other editor things that dont have a save button

https://trello.com/c/5OchsRZO/652-in-wysiwyg-its-confusing-that-you-have-to-click-save-in-the-field-options-in-order-for-them-to-save

**Comment by Mayo19** [comment]
before vs after
screen shot 2018-02-02 at 4 04 27 pm

File: src/editor-actions/Hyperlink.js [hunk 1]
```diff
@@ -113,22 +110,13 @@ export default class Hyperlink extends React.Component {
...
-    });
+    }, this.handleSave);
   }
 
-  handleSave(e) {
-    if (e) {
-      e.preventDefault();
-    }
+  handleSave() {
     const { onChange, onToggleActive } = this.props;
```
Comment (chelliah) [inline src/editor-actions/Hyperlink.js]: you can remove `onToggleActive` here

File: src/editor-actions/ImageOnClick.js [hunk 2]
```diff
@@ -156,14 +152,14 @@ export default class ImageOnClick extends React.Component {
...
-    });
+    }, this.saveAction);
   }
 
   handleFlowIdInput(e) {
     const value = e.target.value;
     this.setState({
       flowId: value
-    });
+    }, this.saveAction);
```
Comment (chelliah) [inline src/editor-actions/ImageOnClick.js]: make sure to remove the `onToggleActive` fns in this saveAction function. like you did in the `Hyperlink` component

File: src/editor-actions/Hyperlink.js (hunk 1 above)
Comment (chelliah) [reply src/editor-actions/Hyperlink.js]: I believe we can remove the 'Button' import at the top of the file too, might be able to do it for a lot of the components the save button is removed from

File: src/editor-actions/SelectionFieldOptions.js [hunk 3]
```diff
@@ -106,26 +103,17 @@ export default class SelectFieldOptions extends React.Component {
...
-    });
+    }, this.handleSave);
   }
 
-  handleSave(e) {
-    if (e) {
-      e.preventDefault();
-    }
+  handleSave() {
     const { localState, persistedState, onChange, onToggleActive } = this.props;
```
Comment (chelliah) [inline src/editor-actions/SelectionFieldOptions.js]: can remove onToggleActive here as well

**Comment by chelliah** [comment]
Awesome! you can remove the 'Button' imports in at least a few of these files, as well as the 'onToggleActive' declarations in the handleSave() fns. ImageOnClick still calls this function, which ends up getting a little buggy. Apart from those few things, this looks good to go!

**Pull request merged**